parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -222,6 +222,19 @@ private static MessageType getParquetTypeWithIds(
     return ParquetSchemaUtil.addFallbackIds(type);
   }
 
+  /**
+   * Returns a list of offsets in ascending order determined by the starting position of the row
+   * groups.
+   */
+  public static List<Long> getSplitOffsets(InputFile file) {
+    try (ParquetFileReader reader = ParquetFileReader.open(ParquetIO.file(file))) {
+      ParquetMetadata md = reader.getFooter();
+      return getSplitOffsets(md);
+    } catch (IOException e) {
+      throw new RuntimeIOException(e, "Failed to read footer of file: %s", file);
```

Comment:
[minor] should we throw `java.io.UncheckedIOException` instead considering we have marked RuntimeIOException as deprecated ?

Reply:
Should we update the other function to also raise the same exception?